modules/vectorfieldvisualizationgl/src/processors/datageneration/vectorfieldgenerator2d.cpp

```diff
@@ -58,51 +47,90 @@ const ProcessorInfo VectorFieldGenerator2D::processorInfo_{
-    Tags::GL,                             // Tags
+    Tags::GL | Tag{"Layer"},              // Tags
+    "Generate a 2D vector field by using separate functions in u and v direction."_unindentHelp,
 };
+
 const ProcessorInfo VectorFieldGenerator2D::getProcessorInfo() const { return processorInfo_; }
 
 VectorFieldGenerator2D::VectorFieldGenerator2D()
     : Processor()
-    , outport_("outport", DataVec2Float32::get(), false)
-    , size_("size", "Field size", size2_t(16), size2_t(1), size2_t(1024))
-    , xRange_("xRange", "X Range", -1, 1, -10, 10)
-    , yRange_("yRange", "Y Range", -1, 1, -10, 10)
-    , xValue_("x", "X", "-x", InvalidationLevel::InvalidResources)
-    , yValue_("y", "Y", "y", InvalidationLevel::InvalidResources)
+    , outport_{"outport"}
+    , size_{"size", "Field size",
+            util::ordinalCount(ivec2{16}, ivec2{1024}).set("Dimensions of the vector field"_help)}
+    , domainU_("xRange", "U Domain",
+               "Domain in u direction for which the functions are evaluated."_help, -1.0f, 1.0f,
+               -10.0f, 10.0f)
+    , domainV_("yRange", "V Domain",
+               "Domain in v direction for which the functions are evaluated."_help, -1.0f, 1.0f,
+               -10.0f, 10.0f)
+    , xValue_("x", "U",
+              "Function u(x,y) evaluated along u direction (uses GLSL functionality)."_help, "-x",
+              InvalidationLevel::InvalidResources)
+    , yValue_("y", "V",
+              "Function v(x,y) evaluated along v direction (uses GLSL functionality)."_help, "y",
+              InvalidationLevel::InvalidResources)
+    , information_("Information", "Data Information")
+    , basis_("Basis", "Basis and Offset")
     , shader_("vectorfieldgenerator2d.frag", Shader::Build::No) {
+
     addPort(outport_);
 
-    addProperties(size_, xValue_, yValue_, xRange_, yRange_);
+    domainU_.setSemantics(PropertySemantics::Text);
+    domainV_.setSemantics(PropertySemantics::Text);
+    addProperties(size_, xValue_, yValue_, domainU_, domainV_, information_, basis_);
+
+    setAllPropertiesCurrentStateAsDefault();
 }
 
 VectorFieldGenerator2D::~VectorFieldGenerator2D() = default;
 
 void VectorFieldGenerator2D::initializeResources() {
-    shader_.getFragmentShaderObject()->addShaderDefine("X_VALUE(x,y)", xValue_.get());
-    shader_.getFragmentShaderObject()->addShaderDefine("Y_VALUE(x,y)", yValue_.get());
+    shader_.getFragmentShaderObject()->addShaderDefine("U_VALUE(x,y)", xValue_.get());
+    shader_.getFragmentShaderObject()->addShaderDefine("V_VALUE(x,y)", yValue_.get());
 
     shader_.build();
 }
 
 void VectorFieldGenerator2D::process() {
+    bool reattach = false;
+    const LayerConfig newConfig{.dimensions = size_.get(),
+                                .format = DataVec2Float32::get(),
+                                .swizzleMask = swizzlemasks::defaultData(0)};
+    if (!layer_ || config != newConfig) {
+        config = newConfig;
+        layer_ = std::make_shared<Layer>(config);
```

Comment:
this could use the same cache technique as in LayerGLProcessor

Reply:
future PR